packages/data-stores/src/plans/selectors.ts

```diff
@@ -131,26 +132,28 @@ export const getPlanByPath = (
 
 	return select( STORE_KEY )
 		.getSupportedPlans( locale )
-		.find( ( plan ) => plan.periodAgnosticSlug === planProduct?.periodAgnosticSlug );
+		.find( ( plan ) => plan.periodAgnosticSlug === planProduct.periodAgnosticSlug );
 };
 
 export const getPlanProduct = (
 	_state: State,
-	periodAgnosticSlug: string | undefined,
-	billingPeriod: PlanProduct[ 'billingPeriod' ] | undefined
+	periodAgnosticSlug?: string,
+	billingPeriod?: PlanProduct[ 'billingPeriod' ]
 ): PlanProduct | undefined => {
 	if ( ! periodAgnosticSlug || ! billingPeriod ) {
 		return undefined;
 	}
-	const products = select( STORE_KEY ).getPlansProducts();
-	if ( periodAgnosticSlug === TIMELESS_PLAN_FREE ) {
-		return products.find( ( product ) => product.periodAgnosticSlug === periodAgnosticSlug );
-	}
-	const planProduct = products.find(
-		( product ) =>
-			product.billingPeriod === billingPeriod && product.periodAgnosticSlug === periodAgnosticSlug
-	);
-	return planProduct;
+
+	return select( STORE_KEY )
+		.getPlansProducts()
+		.find( ( product ) => {
+			const matchesSlug = product.periodAgnosticSlug === periodAgnosticSlug;
+			// The billing period doesn't matter when dealing with free plan
+			const matchesBillingPeriod =
+				periodAgnosticSlug === TIMELESS_PLAN_FREE || product.billingPeriod === billingPeriod;
+
+			return matchesSlug && matchesBillingPeriod;
+		} );
```

Comment:
Refactored the code without changing its logic.
The new code has less branches and explains how the `billingPeriod` doesn't affect the free plan